spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GoogleCloudContextAutoConfiguration.java

```diff
@@ -14,26 +17,26 @@
 
 /**
  * @author Vinicius Carvalho
+ *
+ * Base starter for Google Cloud Projects. Provide defaults for {@link GoogleCredentials}.
+ * Binds properties from {@link org.springframework.cloud.gcp.core.GoogleCloudProperties}
+ *
  */
 @Configuration
 @ConditionalOnClass(GoogleCredentials.class)
-public class CredentialsAutoConfiguration implements EnvironmentAware {
+@EnableConfigurationProperties(GoogleCloudProperties.class)
+public class GoogleCloudContextAutoConfiguration {
```

Comment:
Naming is not a big issue, and it's the most useless waste of anyone's effort going on this. That is the very single thing that will take me off my base if we need to start having discussions around, because *NO ONE* gets naming right.

A few things on the naming:

- Usually the AutoConfiguration is related to some feature you want to add, We are not adding `Properties` on the auto configuration module, `Properties` are a feature that those modules can use to abstract variables from the system (environment, config files, config server, you name it). So `XXXPropertiesAutoConfiguration` doesn't make much sense.

- So the question is are we using `GoogleCloud` or `Gcp` which is the same as debate `Tomato or Tomato`. If the others think that Gcp is preferred than `GoogleCloud` by no means I'll make a problem out of it.

 I'd prefer to use GoogleCloud to be more specific about what this is about. Gcp could akso mean `Great Circus Parade` and we don't want that to confuse our users